lib/Conversion/ATenToTCF/CoreOpConversionPatterns.cpp

```diff
@@ -66,6 +66,19 @@ class ConvertBinaryElementwise : public OpRewritePattern<SourceOp> {
   }
 };
 
+/// The ATen Conv2dOp has seven arguments:
+///   input, weight, bias, stride, padding, dilation, groups
+class ConvertATenConv2d : public OpRewritePattern<aten::Conv2dOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(aten::Conv2dOp srcConv2dOp,
+                                PatternRewriter &rewriter) const override {
+    rewriter.replaceOpWithNewOp<tcf::ConvNCHWOp>(
```

Comment:
check that the stride/padding/dilation are the default values to ensure you don't miscompile

Reply:
I am struggling with MLIR's Matchers API. Functions such as `m_ConstantInt()` don't match NPCOMP's `basicpy.ListType`, which defines stride/padding/dilation in this PR's `npcomp-run-mlir` test. Debugger reveals that `basicpy.ListType` is neither vector- nor tensor-like. There are other ways to solve constraining said inputs, but what direction would you take here?

Reply:
You can just check `getDefiningOp<BuildListOp>` or whatever and manually check the operands.